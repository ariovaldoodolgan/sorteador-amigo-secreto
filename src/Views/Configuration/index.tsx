import Card from "../../components/Card";
import Footer from "../../components/Footer";
import Form from "../../components/Form";
import ParticipantList from "../../components/ParticipantList";

const Configuration = () => {

    return (
        <Card>
            <section>
                <h2>Let's start!</h2>
                <Form />
                <ParticipantList />
                <Footer />
            </section>
        </Card>
    )
}

export default Configuration;