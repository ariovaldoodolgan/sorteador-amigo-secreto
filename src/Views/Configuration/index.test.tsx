import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Configuration from ".";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {

    return {
        useNavigate: () => mockNavigate
    }
})

describe('Configuration page', () => {

    test('Should be rendered correctly', () => {
        const { container } = render(
            <RecoilRoot>
                <Configuration />
            </RecoilRoot>
        )

        expect(container).toMatchSnapshot()
    })
})