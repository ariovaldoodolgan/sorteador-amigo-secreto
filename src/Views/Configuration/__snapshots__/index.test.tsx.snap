// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration page Should be rendered correctly 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Let's start!
      </h2>
      <form>
        <div
          class="group-input-btn"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Add
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="footer-configurations"
      >
        <button
          class="button"
          disabled=""
        >
          Start the draw
        </button>
        <img
          alt="Shopping bag"
          src="/images/bags.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
